### PR TITLE
After hub recovery, cant initiate failover

### PR DESCRIPTION
**Pull request by GowthamShanmugam**
BZ: https://bugzilla.redhat.com/show_bug.cgi?id=2209288

### CODE REVIEW

File: packages/mco/utils/acm.ts [hunk 1]
```diff
@@ -65,3 +74,23 @@ export const getClustersFromDecisions = (
...
+) =>
+  getAnnotations(drPlacementControl)?.[
+    LAST_APP_DEPLOYMENT_CLUSTER_ANNOTATION
+  ] || '';
+
+export const findDeploymentClusters = (
+  placementDecision: ACMPlacementDecisionKind,
+  drPlacementControl: DRPlacementControlKind
+): string[] => {
+  if (placementDecision?.status.decisions.length > 0) {
```
Comment (SanjalKatiyar) [inline packages/mco/utils/acm.ts:89]: do we need to do this or its guaranteed that if "status" is present then "decisions" will be present as well ??
```suggestion
  if ((placementDecision ?? {}).status?.decisions?.length > 0) {
```

**Comment by SanjalKatiyar** [comment]
so, this annotation will never get removed ?? not even during hub recovery ??

**Comment by GowthamShanmugam** [comment]
> so, this annotation will never get removed ?? not even during hub recovery ??

When ramen starts to reconcile, it will add the annotation immediately. In hub recovery they will get this info from s3 object storage and display it as a last remembered application cluster.

File: packages/mco/utils/acm.ts (hunk 1 above)
Comment (GowthamShanmugam) [reply packages/mco/utils/acm.ts:89]: ack

**Pull request merged**